thorlcr/graph/thgraph.cpp

```diff
@@ -2488,6 +2488,14 @@ CJobBase::CJobBase(const char *_graphName) : graphName(_graphName)
     jobSlaveChannelNum.allocateN(querySlaves()); // filled when channels are added.
     for (unsigned s=0; s<querySlaves(); s++)
         jobSlaveChannelNum[s] = NotFound;
+    StringBuffer wuXML;
+    if (!getEmbeddedWorkUnitXML(querySo, wuXML))
+        throwUnexpected();
```

Comment:
getEmbeddedWorkUnitXML() already throws an exception if !loadCurrentExecutable(), so probably impossible to return NULL.